### PR TITLE
Dev pathlib switch and pathwork overhaul

### PR DESCRIPTION
**Pull request by madeddy**
Tested all and works here.
I changed also unrpyc-compile, compiled un.rpyc and it worked AFAIK but this should be double checked.

See what you think. ✌🏻

### CODE REVIEW

File: un.rpyc/unrpyc-compile.py [hunk 1]
```diff
@@ -31,41 +31,39 @@ def read_ast_from_file(raw_contents):
...
 
-    logfile = path.join(os.getcwd(), "game/unrpyc.log.txt")
+    logfile = Path.cwd().joinpath("game/unrpyc.log.txt")
     ensure_dir(logfile)
-    with open(logfile, "w") as f:
+    with logfile.open("w") as f:
         f.write("Beginning decompiling\n")
 
-        for abspath, fn, dir, data in sys.files:
+        for fullpath, fn, dir_name, data in sys.files:
```
[the remark this answers is not in the document]
Comment (madeddy) [reply un.rpyc/unrpyc-compile.py:57]: > Do I need to make those?

Yes, please do!
If i touch un.rpyc stuff you should always see it as unsure. I trust myself somewhat with this module, but not with compile etc.

Hope the renaming is OK as this are taken python names.

File: un.rpyc/unrpyc-compile.py [hunk 2]
```diff
@@ -31,41 +31,39 @@ def read_ast_from_file(raw_contents):
     return stmts
 
 def ensure_dir(filename):
-    dir = path.dirname(filename)
-    if dir and not path.exists(dir):
-        os.makedirs(dir)
+    dir_name = Path(filename).parent
+    if dir_name and not dir_name.exists():
+        dir_name.makedir()
```
[the remark this answers is not in the document]
Comment (madeddy) [reply un.rpyc/unrpyc-compile.py:36]: Correct, overlooked this.

[the remark this answers is not in the document]
Comment (CensoredUsername) [reply un.rpyc/unrpyc-compile.py]: damnit, I always mess that one up.

[the remark this answers is not in the document]
Comment (madeddy) [reply un.rpyc/unrpyc-compile.py]: I wondered how i overlooked makedirs as i am sure this happens not the first time to me. So, i am not the only one.

File: un.rpyc/unrpyc-compile.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (CensoredUsername) [reply un.rpyc/unrpyc-compile.py]: >If i touch un.rpyc stuff you should always see it as unsure. I trust myself somewhat with this module, but not with compile etc.

I ended up thinking a bit about this, and realized that providing some easier testing than the manual "insert into ren'py game, see results" would be nice. So now there's additional automated testing so you can see if you break something there, and you at least get a call trace if anything breaks. And I see it in CI when I need to fix something there.

[the remark this answers is not in the document]
Comment (madeddy) [reply un.rpyc/unrpyc-compile.py]: Very good. Thanks.

**Pull request merged**